pkg/util/errors.go

```diff
@@ -186,3 +192,145 @@ func HandleStorageError(err error, entityName string) error {
 		}
 	}
 }
+
+func HandleReferencesError(err error, guidsArray []string) error {
+	if err == nil {
+		return nil
+	}
+
+	if _, ok := err.(*HTTPError); ok {
+		return err
+	}
+
+	switch err {
+	case ErrSharedInstanceHasReferences:
+		errorMessage := fmt.Sprintf("Couldn't delete the service instance. Before you can delete it, you first need to delete these %d references: %s", len(guidsArray), guidsArray)
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: errorMessage,
+			StatusCode:  http.StatusBadRequest,
+		}
+	case ErrUnsharingInstanceWithReferences:
+		errorMessage := fmt.Sprintf("Couldn't unshare the service instance. Before you can unshare it, you first need to delete these %d references: %s", len(guidsArray), guidsArray)
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: errorMessage,
+			StatusCode:  http.StatusBadRequest,
+		}
+	default:
+		return err
+	}
+}
+
+var (
+	ErrCatalogUsesReservedPlanName           = errors.New("catalog contains a reserved plan name")
+	ErrPlanMustBeBindable                    = errors.New("plan must be bindable")
+	ErrAsyncNotSupportedForSharing           = errors.New("can't use async for sharing instances")
+	ErrReferencedInstanceNotShared           = errors.New("referenced-instance should be shared first")
+	ErrChangingPlanOfReferenceInstance       = errors.New("changing plan of reference instance")
+	ErrChangingPlanOfSharedInstance          = errors.New("changing plan of shared instance")
+	ErrChangingParametersOfReferenceInstance = errors.New("changing parameters of reference instance")
+	ErrMissingOrInvalidReferenceParameter    = errors.New("missing or invalid referenced_instance_id parameter")
+	ErrParsingNewCatalogWithReference        = errors.New("failed generating reference-plan")
+	ErrUnknownOSBMethod                      = errors.New("osb method is unknown")
+	ErrSharedPlanHasReferences               = errors.New("shared plan has references")
+	ErrInstanceIsAlreadyAtDesiredSharedState = errors.New("instance already at the desired shared state")
+	ErrInvalidShareRequest                   = errors.New("invalid share request")
+)
+
+func HandleInstanceSharingError(err error, entityName string) error {
+	if err == nil {
+		return nil
+	}
+
+	if _, ok := err.(*HTTPError); ok {
+		return err
+	}
+
+	switch err {
+	case ErrCatalogUsesReservedPlanName:
+		return &HTTPError{
+			ErrorType:   "BadRequest",
+			Description: fmt.Sprintf("The plan name you used for registration \"%s\" is reserved for the Service Manager; you must choose a different name.", entityName),
```

Comment:
why "reserved for *service manager*"? simply use "is a reserved name"

Reply:
"The plan name you used for registration \"%s\" is a reserved name; you must choose a different name.", entityName)

Reply:
changed to `"The plan name you used for registration \"%s\" is a reserved name; you must choose a different name."`